doc/specs/#885 - Terminal Settings Model.md

```diff
@@ -78,13 +78,126 @@ This separation leaves `AppKeyBindings` with the responsibility of detecting and
  `KeyMapping` handles the (de)serialization and navigation of the key bindings.
 
 
+### Fallback Value
+
+Cascading settings allows our settings model to be constructed in layers (i.e. settings.json values override defaults.json values). With the upcoming introduction of the Settings UI and serialization, it is important to know where a setting value comes from. Consider a Settings UI displaying the following information:
+```json
+    // <profile>: <color scheme value>
+    "defaults": "Solarized", // profiles.defaults
+    "A": "Raspberry", // profile A
+    "B": "Tango", // profile B
+    "C": "Solarized" // profile C
+```
+If `profiles.defaults` gets changed to `"Tango"` via the Settings UI, it is unclear if profile C's value should be updated as well. We need profile C to record if it's value is inherited from profile.defaults or explicitly set by the user.
+
+#### Object Model Inheritance
+
+To start, each settings object will now have a `CreateChild()` function. For `GlobalAppSettings`, it will look something like this:
+```c++
+GlobalAppSettings GlobalAppSettings::CreateChild() const
+{
+    GlobalAppSettings child {};
+    child._parent = this;
+    return child;
+}
+```
+`_parent` serves as a reference for who to ask if a settings value was not provided by the user. `LaunchMode`, for example, will now have a getter/setter that looks similar to this:
+```c++
+// returns the resolved value for this setting
+LaunchMode GlobalAppSettings::LaunchMode()
+{
+    // fallback tree:
+    //  - user set value
+    //  - inherited value
+    //  - system set value
+    return til::coalesce_value(_LaunchMode, _parent.LaunchMode(), LaunchMode::DefaultMode);
+}
+
+void GlobalAppSettings::LaunchMode(IReference<LaunchMode> val)
+{
+    // if val is nullptr, we are explicitly saying that we want the inherited value
+    _LaunchMode = val;
+}
+
+// returns the user set value for this setting
+// NOTE: This is important for the Settings UI to identify whether the user explicitly or implicitly set the presented value
+LaunchMode GlobalAppSettings::ExplicitLaunchMode()
+{
+    return _LaunchMode;
+}
+```
+
+Additionally, each settings object will now have an `ApplyTo()` function. For `Profile`, it will look something like this:
+```c++
+// layers the Profile settings onto another profile
+void Profile::ApplyTo(Profile profile) const
+{
+    if (_fontSize)
+    {
+        FontSize(_fontSize);
+    }
+
+    // repeat for all settings
+}
+```

```

Comment:
As mentioned in review, this needs some `profile.`'s scattered throughout to make it clear who is applying to who